raft-tech/TANF-app
# 2133 tribe discoverability

## Summary of Changes
Pull request closes #2133

## How to Test
```
cd tdrs-backend && docker-compose up
cd tdrs-frontend && docker-compose up --build
```

1. Open http://localhost:3000/ and sign in. Make sure your user is in the `Initial` access request state and has no location or roles assigned.
2. Navigate to `/home` and fill out the request access form
3. Select a jurisdiction type, inspect how the stt selection box updates based on the selected type.
   * Note, if you have selected a jurisdiction type, selected an stt for that jurisdiction, and then change your jurisdiction type, the select-box does not clear out. You have to click into the select-box, then click out to clear the selection. This [is a bug in uswds](https://github.com/uswds/uswds/discussions/4391). The logic to clear the selection on jurisdiction type change has been added, so we should see it work immediately upon upgrade if this is fixed in an later version of uswds.
4. Complete access request

## Deliverables

Checklist of ACs:
+ [x] Autocomplete is NOT be enabled on the STT combobox field
+ [x] STT combobox appears after jurisdiction type radio button selection
+ [x] STT combobox is dynamically labeled based on radio selection
+ [x] Combobox options are limited to the scope of the radio selection
+ [x] Form errors reflect the number of visible errors (e.g. error count should not read 3 and not 4 if a user attempts to submit the form without choosing a jurisdiction type
+ [x] Keyboard navigable. Test via [activating a screenreader](https://hackmd.io/Z5t247a7St2tZzLS35JBkw) and navigating the form with keyboard only. The below should be default behavior but it's worth explicitly testing on a dynamic form.
    + [x] The STT combobox shouldn't be tabbable before it's visible
    + [x] Once a radio selection is made and the combobox appears it should be the next tab stop after the radio select.

### [Deliverable 2: Tested Code](https://github.com/raft-tech/TANF-app/blob/develop/docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-2-Tested-Code)

+ Are all areas of code introduced in this PR meaningfully tested?
+ Are code coverage minimums met?

### [Deliverable 3: Properly Styled Code](https://github.com/raft-tech/TANF-app/blob/develop/docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-3-Properly-Styled-Code)


### [Deliverable 4: Accessible](https://github.com/raft-tech/TANF-app/blob/develop/docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-4-Accessibility)



### [Deliverable 5: Deployed](https://github.com/raft-tech/TANF-app/blob/develop/docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-5-Deployed)


### [Deliverable 6: Documented](https://github.com/raft-tech/TANF-app/blob/develop/docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-6-Code-documentation)


### [Deliverable 7: Secure](https://github.com/raft-tech/TANF-app/blob/develop/docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-7-Secure)


### [Deliverable 8: User Research](https://github.com/raft-tech/TANF-app/blob/develop/docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-8-User-Research)

Research product(s) clearly articulate(s):
